File: src/main/java/models/ForumReport.java
package models;

public class ForumReport {
    private int id;
    private String type;
    private int targetId;
    private int reporterId;
    private String reporterName;
    private String targetContent;
    private String reason;
    private String status;
    private String createdAt;

    public ForumReport() {}

    public ForumReport(int id, String type, int targetId, int reporterId, String reporterName, 
                   String targetContent, String reason, String status, String createdAt) {
        this.id = id;
        this.type = type;
        this.targetId = targetId;
        this.reporterId = reporterId;
        this.reporterName = reporterName;
        this.targetContent = targetContent;
        this.reason = reason;
        this.status = status;
        this.createdAt = createdAt;
    }

    // Getters
    public int getId() { return id; }
    public String getType() { return type; }
    public int getTargetId() { return targetId; }
    public int getReporterId() { return reporterId; }
    public String getReporterName() { return reporterName; }
    public String getTargetContent() { return targetContent; }
    public String getReason() { return reason; }
    public String getStatus() { return status; }
    public String getCreatedAt() { return createdAt; }

    // Setters
    public void setId(int id) { this.id = id; }
    public void setType(String type) { this.type = type; }
    public void setTargetId(int targetId) { this.targetId = targetId; }
    public void setReporterId(int reporterId) { this.reporterId = reporterId; }
    public void setReporterName(String reporterName) { this.reporterName = reporterName; }
    public void setTargetContent(String targetContent) { this.targetContent = targetContent; }
    public void setReason(String reason) { this.reason = reason; }
    public void setStatus(String status) { this.status = status; }
    public void setCreatedAt(String createdAt) { this.createdAt = createdAt; }
}

File: src/Entity/ForumReport.php
<?php

namespace App\Entity;

use App\Repository\ForumReportRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ForumReportRepository::class)]
#[ORM\Table(name: 'forum_report')]
class ForumReport
{
    public const STATUS_PENDING = 'pending';
    public const STATUS_REVIEWED = 'reviewed';
    public const STATUS_DISMISSED = 'dismissed';

    public const TYPE_DISCUSSION = 'discussion';
    public const TYPE_MESSAGE = 'message';

    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 20)]
    private ?string $type = null;

    #[ORM\Column(name: 'target_id', type: 'integer')]
    private ?int $targetId = null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'reporter_id', referencedColumnName: 'id', nullable: false)]
    private ?User $reporter = null;

    #[ORM\Column(length: 255)]
    private ?string $reason = null;

    #[ORM\Column(length: 20, options: ['default' => 'pending'])]
    private ?string $status = null;

    #[ORM\Column(name: 'created_at', type: 'datetime')]
    private ?\DateTimeInterface $createdAt = null;

    public function __construct()
    {
        $this->createdAt = new \DateTime();
        $this->status = self::STATUS_PENDING;
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): static
    {
        if (!in_array($type, [self::TYPE_DISCUSSION, self::TYPE_MESSAGE])) {
            throw new \InvalidArgumentException('Invalid report type');
        }
        $this->type = $type;
        return $this;
    }

    public function getTargetId(): ?int
    {
        return $this->targetId;
    }

    public function setTargetId(int $targetId): static
    {
        $this->targetId = $targetId;
        return $this;
    }

    public function getReporter(): ?User
    {
        return $this->reporter;
    }

    public function setReporter(?User $reporter): static
    {
        $this->reporter = $reporter;
        return $this;
    }

    public function getReason(): ?string
    {
        return $this->reason;
    }

    public function setReason(string $reason): static
    {
        $this->reason = $reason;
        return $this;
    }

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(string $status): static
    {
        if (!in_array($status, [self::STATUS_PENDING, self::STATUS_REVIEWED, self::STATUS_DISMISSED])) {
            throw new \InvalidArgumentException('Invalid status');
        }
        $this->status = $status;
        return $this;
    }

    public function getCreatedAt(): ?\DateTimeInterface
    {
        return $this->createdAt;
    }

    public function setCreatedAt(\DateTimeInterface $createdAt): static
    {
        $this->createdAt = $createdAt;
        return $this;
    }

    public function isPending(): bool
    {
        return $this->status === self::STATUS_PENDING;
    }
}
